### ISSUE TITLE
"create_stoichiometric_array" is not very intuitive

### ISSUE THREAD

**Issue by joaocardoso** [opencobra/cobrapy#442]
Hi!

I was made aware of `cobra.util.array.create_stoichiometric_array`.
I don't think it is the best naming. The stoichiometric matrix (S), is part of the conceptual representation of the constraint-based model. 

Anyone who learns about these models, learns first about how to build a S matrix and then how to convert it into a linear problem. As a user, I will be looking for the stoichiometric matrix in the documentation. I suggest renaming the function to `create_stoichiometric_matrix`.

Cheers,

João

**Comment by pstjohn** [comment]
We talked about having this function be called by a model property, `model.S`, but this was a controversial issue 😄 . It got removed in #425 i believe. 

Is your problem with the function the fact that it's buried in the cobrapy util namespace? Or that it's called ...`_array` rather than ...`_matrix`?

*Note:* One of the reasons for `array` is that we return a numpy array, (rather than a numpy matrix, which caused some headaches in previous versions). Now that pandas is a hard dependency, it might make sense to return a dataframe, with rows/columns labelled with metabolite id's.
